doc/side-by-side-CTL-and-Plutus-comparison.md

```diff
@@ -0,0 +1,323 @@
+
+newtype Contract w (s :: Row *) e a = Contract { unContract :: Eff (ContractEffs w e) a }
+  deriving newtype (Functor, Applicative, Monad)
+```
+
+The Plutus `Contract` environment is specialized to just two values and is fixed. 
+Also, Plutus `Contract` uses a phantom type `s` to contract schema 
+and parameters `w` for a writer and `e` for errors. 
+In the case of CTL we don't have the contract schema parameter or the writer 
+parameter since CTL definition allows performing arbitrary effects. 
+This is possible  since the definition of `LoggerT` is: 
+
+```PureScript
+newtype LoggerT m a = LoggerT ((Message -> m Unit) -> m a)
+```
+
+The use of `Aff` inside `LoggerT` allows us to use asynchronous effects 
+inside the logger. In particular, this has a similar effect as using `IO` 
+in Haskell, although isn't the same.
+
+
+## Contract comparison
+
+We can now begin to compare contracts. 
+
+The most famous contracts are those contained as part of 
+the [Plutus pioneer program](https://plutus-pioneer-program.readthedocs.io/en/latest/pioneer/week2.html) in week2. Both of them use the same on-chain contract that 
+allows an arbitrary datum and arbitrary redeemer. 
+
+### Signature and submit 
+
+We are going to need the auxiliary PureScript function that isn't part 
+of CTL: 
+
+```PureScript
+buildBalanceSignAndSubmitTx
+  :: Lookups.ScriptLookups PlutusData
+  -> TxConstraints Unit Unit
+  -> Contract () TransactionHash
+buildBalanceSignAndSubmitTx lookups constraints = do
+  ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+  BalancedSignedTransaction bsTx <-
+    liftedM "Failed to balance/sign tx" $ balanceAndSignTx ubTx
+  txId <- submit bsTx.signedTxCbor
+  logInfo' $ "Tx ID: " <> show txId
+  pure txId
+```
+
+
+This function takes our lookups and constraints, then constructs
+an unbalanced transaction using them, after that it tries to balance 
+the transaction and attempts to get a signature for this transaction, then it 
+submits the transaction and if all got right, it returns the Id
+for the transaction wrapped inside the `Contract`. 
+
+This is a separate function as is almost boilerplate once we 
+have set our environment.
+
+We can see this function as a more explicit version of the 
+process made by Plutus with the `submitTx` function: 
+
+```Haskell 
+-- | Build a transaction that satisfies the constraints, then submit it to 
+-- | the network. The constraints do not refer to any 
+-- | typed script inputs or outputs.
+submitTx 
+  :: forall w s e. AsContractError e 
+    => TxConstraints Void Void -> Contract w s e CardanoTx
+```
+
+A major difference in both functions is the fact that we 
+are also signing the transaction as part of the balance of 
+the transaction.
+
+
+### MustPayTo functions
+
+
+In the case of Plutus `Contract`, we use the function 
+`mustPayToOtherScript`, according to Plutus ledger, it :
+
+```Haskell
+-- | mustPayToOtherScript vh d v locks the value v with the given script 
```

Comment:
```suggestion
-- | `mustPayToOtherScript vh d v` locks the value v with the given script 
```

Please also backtick-quote all variable names.

Reply:
I have my doubts here since I now copying the comments from the source and it uses the haddock style:

`@mustPayToOtherScript vh d v@`

Reply:
Purescript docs use markdown.

Reply:
this part is Haskell code.